v3/src/components/data-display/models/data-configuration-model.ts

```diff
@@ -354,8 +334,120 @@ export const DataConfigurationModel = types
             numRepetitions = Math.max(this.categoryArrayForAttrRole('topSplit').length, 1)
         }
         return numRepetitions
+      },
+      hasSingleSubplot() {
+        // A graph has a single subplot if it has one or fewer categorical attributes, or if it has exactly two
+        // categorical attributes on axes that are perpendicular to each other.
+        const attrTypes = {
+          bottom: self.attributeType("x"),
+          left: self.attributeType("y"),
+          top: self.attributeType("topSplit"),
+          right: self.attributeType("rightSplit")
+        }
+        const xHasCategorical = attrTypes.bottom === "categorical" || attrTypes.top === "categorical"
+        const yHasCategorical = attrTypes.left === "categorical" || attrTypes.right === "categorical"
+        const categoricalAttrCount = Object.values(attrTypes).filter(a => a === "categorical").length
+        const hasOnlyTwoCategorical = categoricalAttrCount === 2
+
+        return categoricalAttrCount <= 1 || (hasOnlyTwoCategorical && xHasCategorical && yHasCategorical)
       }
-    }))
+  }))
+  .views(self => ({
+    isCaseInSubplot(subPlotKey: Record<string, string>, caseData: Record<string, any>) {
+      const topAttrID = self.attributeID("topSplit")
+      const rightAttrID = self.attributeID("rightSplit")
+      const isSplitMatch = (!topAttrID || (topAttrID && subPlotKey[topAttrID] === caseData[topAttrID])) &&
+        (!rightAttrID || (rightAttrID && subPlotKey[rightAttrID] === caseData[rightAttrID]))
+      
+      return self.hasSingleSubplot() || isSplitMatch
+    },
+    isCaseInCell(subPlotKey: Record<string, string>, caseData: Record<string, any>) {
+      const numOfKeys = Object.keys(subPlotKey).length
+      let matchedValCount = 0
+      Object.keys(subPlotKey).forEach(key => {
+        if (subPlotKey[key] === caseData[key]) matchedValCount++
+      })
+      return matchedValCount === numOfKeys
+    }
+  }))
+  .views(self => ({
+    subPlotCases(subPlotKey: Record<string, string>) {
+      const casesInPlot = [] as ICase[]
```

Comment:
In the first `forEach` below, when there are multiple numeric y-attributes, cases will (usually) be counted more than once. I've written a PT story (#185854897) about this, a fix of which should probably be in a separate PR. This comment applies to each of the following three methods as well.